napari/_qt/layers/qt_image_base_layer.py

```diff
@@ -28,16 +28,13 @@ def __init__(self, layer):
 
         # Create contrast_limits slider
         self.contrastLimitsSlider = QHRangeSlider(
-            slider_range=[0, 1, 0.0001], values=[0, 1]
+            self.layer.contrast_limits, self.layer._contrast_limits_range
         )
-        self.contrastLimitsSlider.setEmitWhileMoving(True)
-        self.contrastLimitsSlider.collapsable = False
-        self.contrastLimitsSlider.setEnabled(True)
 
-        self.contrastLimitsSlider.rangeChanged.connect(
-            self.contrast_limits_slider_changed
-        )
-        self.contrast_limits_slider_update()
+        set_clim = partial(setattr, self.layer, 'contrast_limits')
```

Comment:
Is there a reason to use the `partial` syntax here? Should we be using it elsewhere too? I'm not so familiar with it (had to do some googling to read more!!)

Reply:
I'm quite, ahem, partial to it myself! =P It's standard Python but for some reason (partly because Guido van Rossum doesn't much like functional programming, I'm sure), it's not very common.

> Should we be using it elsewhere

I don't think it's mandatory or worth homogenising, but I personally like it. My preference would actually be to use [`toolz.curry`](https://toolz.readthedocs.io/en/latest/curry.html#id1) on everything, but that is *non-*standard, so I restrain myself. =) I have been toying with the idea of authoring a PEP for this.

Reply:
no reason other than personal preference.  i find myself reaching for it a lot with Qt signals, because it pains me to do something like this:
```python
def setter(x):
    self.something = x

widget.signal.connect(setter)
```
(and also, here, there were multiple setters to declare)